src/curve/constant-product/ConstantProductPair.sol

```diff
@@ -86,8 +86,6 @@ contract ConstantProductPair is ReservoirPair {
                     }
                 }
             }
-        } else if (kLast != 0) {
-            kLast = 0;
```

Comment:
Interesting, this was actually a bug but they didn't flag it as such. We should keep this change but also remove the return value `bool rFeeOn`.

Additionally, I think we can remove `if (lSqrtOldK != 0)` but lets do that later so we can add an issue now tracking it

Reply:
This is also the case for Stablepair i believe

Reply:
alright so after reading your comment, I thought we still needed this check because in `_calcFee`, it's dividing this by this number so if we remove the check we're going to have a division by zero error. E.g. for first mints. 

But I tested it out and it doesn't happen in both curves, due to the `lSqrtNewK > lSqrtOldK` check. In the first mint everything is zero so this problem wouldn't be triggered. 

Created an issue to track this